### PR TITLE
update maintainer rules

### PR DESCRIPTION
**Pull request by NicolasMassart**
## PR description

These changes are expected to clarify the rules to reach maintainer status. Some edits were made on the voting process, template PR description added, simplified language,...

The PR also adds a new maintainer type in addition to code maintainer to enable people who significantly contribute to the Besu project to be able to access some project management features such as issues triage and assignation.
Some features, expecially when using Zenhub, indeed require a write access on the repository even though the contributor will not make any code changes. But things like assigning a user to an issue requires maintainer role.

See https://github.com/NicolasMassart/besu/blob/maintainer-rules/MAINTAINERS.md for the easier to read rendered doc.

## Approval of this PR

As for the maintainer addition, this PR being a significant change to the process, I require the same rules to be followed:

- [ ] 2 weeks voting period (ends June 3rd 2021)
- [x] at least 3 approvals
- [x] no veto
- [x] absolute majority makes it pass immediately (currently 11 approvals).

## Fixed Issue(s)

We need significant and fully involved non-code contributors to be able to fully help us on the project without the need to contribute code to become a maintainer.

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/fees/TransactionGasBudgetCalculator.java [hunk 1]
```diff
@@ -30,11 +30,7 @@ static TransactionGasBudgetCalculator frontier() {
   }
 
   static TransactionGasBudgetCalculator eip1559(final EIP1559 eip1559) {
-    return gasBudgetCalculator(
```
Comment (shemnon) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/fees/TransactionGasBudgetCalculator.java:33]: Please move to a separate PR

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java [hunk 2]
```diff
@@ -47,7 +47,6 @@
 import java.util.Map;
 import java.util.NavigableSet;
 import java.util.NoSuchElementException;
-import java.util.Objects;
```
Comment (shemnon) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java:50]: Please move to a separate PR

**Review by shemnon — CHANGES_REQUESTED** [review]
This policy change should only impact markdown.  Please remove the java changes (the master force push may have caused this)

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/fees/TransactionGasBudgetCalculator.java (hunk 1 above)
Comment (NicolasMassart) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/fees/TransactionGasBudgetCalculator.java]: it does only impact MD, only a git issue linked to the previous DCO breaking commit, fixed.

**Review by atoulme — APPROVED** [review]
LGTM

**Review by garyschulte — APPROVED** [review]
:+1:

**Review by matkt — APPROVED** [review]

**Review by ajsutton — APPROVED** [review]

File: MAINTAINERS.md [hunk 3]
```diff
@@ -40,39 +40,67 @@
...
-maintainer. To become a maintainer the following steps occur, roughly in order.
+Besu welcomes community contribution.
+Each community member may progress to become a maintainer.
 
-- 5 significant changes have been authored by the proposed maintainer and
-  accepted.
+There are two ways to become a maintainer:
+
+- Contribute significantly to the code in this repository.
+- Contribute significantly to the doc or being recognised as great help by other contributors.
```
Comment (macfarla) [inline MAINTAINERS.md:49]: s/recognised/recognized

**Review by macfarla — COMMENTED** [review]
one nitpick, otherwise LGTM

**Review by usmansaleem — APPROVED** [review]
LGTM

**Review by jframe — APPROVED** [review]

**Review by lucassaldanha — APPROVED** [review]

**Review by pinges — APPROVED** [review]

**Review by shemnon — APPROVED** [review]
Assuming @NicolasMassart is voting in favor, my vote makes it 11.

**Comment by NicolasMassart** [comment]
I vote in favour but can't approve my own PR, so absolute majority, no veto, I will merge.
Thanks all!
